src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1656,6 +1657,24 @@ private static void ValidateWin32Settings(string win32ResourceFile, string win32
             }
         }
 
+        internal static IEnumerable<InstrumentationKind> ParseInstrumentationKinds(string value, IList<Diagnostic> diagnostics)
```

Comment:
Consider passing in the `ArrayBuilder<InstrumentationKind>` rather than returning `IEnumerable<InstrumentationKind>`. That would avoid an iterator class and would be consistent with `diagnostics` which is an argument.